packages/material-ui/src/Skeleton/Skeleton.js

```diff
@@ -1,31 +1,93 @@
-import * as React from 'react';
+import { keyframes } from '@material-ui/styled-engine';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import { deepmerge } from '@material-ui/utils';
 import clsx from 'clsx';
 import PropTypes from 'prop-types';
+import * as React from 'react';
 import {
   alpha,
-  withStyles,
-  unstable_toUnitless as toUnitless,
   unstable_getUnit as getUnit,
+  unstable_toUnitless as toUnitless,
   useThemeVariants,
 } from '../styles';
+import experimentalStyled from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
+import { getSkeltonUtilityClass } from './skeltonClasses';
+
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
 
-export const styles = (theme) => {
+  return deepmerge(styles.root || {}, {
+    ...styles[styleProps.variant],
+    ...styles[styleProps.animation],
+    ...(styleProps.hasChildren && styles.withChildren),
+    ...(styleProps.hasChildren && !styleProps.width && styles.fitContent),
+    ...(styleProps.hasChildren && !styleProps.height && styles.heightAuto),
+  });
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { classes, variant, animation, hasChildren, width, height } = styleProps;
+
+  const slots = {
+    root: [
+      'root',
+      variant,
+      animation,
+      hasChildren && 'withChildren',
+      hasChildren && !width && 'fitContent',
+      hasChildren && !height && 'heightAuto',
+    ],
+  };
+
+  return composeClasses(slots, getSkeltonUtilityClass, classes);
+};
+
+const pulseKeyframe = keyframes`
+0% {
+  opacity: 1;
+}
+50% {
+  opacity: 0.4;
+}
+100% {
+  opacity: 1;
+}
+`;
+
+const waveKeyframe = keyframes`
+0% {
+  transform: translateX(-100%);
+}
+50% {
+  // +0.5s of delay between each loop
+  transform: translateX(100%);
+}
+100% {
+  transform: translateX(100%);
+}
+`;
+
+const SkeltonRoot = experimentalStyled(
+  'span',
+  {},
+  { name: 'MuiSkelton', slot: 'Root' },
+  overridesResolver,
+)(({ styleProps, theme }) => {
```

Comment:
These styles needs to be converted to the CSS syntax. Take a look on how it was done for the `CircularProgress` component - https://github.com/mui-org/material-ui/pull/24622

Reply:
I just tried it! fc193fc hope I'm in the right path.

Reply:
Looks much better. Couple of things we need to still fix -  let's just try to avoid the `classes` selector as we are increasing specificity with that. For example:

```
  & .${skeltonClasses.circular} {
     border-radius: 50%;
  }
```

should be transformed to:

```
border-radius: ${({ styleProps }) => stylePRops.circular ? '50%' : undefined }
```

Also I remember that we were destructuring the `animation` shorthand to: `animation-name`, `animation-duration` etc.. The `CircularPropgress` should have examples of both of these requirements.

Reply:
I'm sorry, I couldn't find time to fix this for few days. 

I transformed css style and animation shorthand here 972a375 .

I have two questions.
1. duplicated css filed should be summarized in one place ?
like
```
height: ${({ styleProps }) => {
    if (styleProps.variant === 'text' ) {
      return 'auto'
    }
    if (styleProps.hasChildren && !styleProps.height) {
    return 'auto'  
    }
  }
};
```
I just thought doing this makes the code tough to read, because it's not organized by props.

2. I wonder what the reason behind "destructuring the `animation` shorthand". Is there any issue or discussion ?(I couldn't find one)

Reply:
For 1. I agree, for 2. I think there was some issue with styled-components, but I may be wrong. Would be great for this to be tested with both emotion and sc before merge. Cc @oliviertassinari

Reply:
I don't understand why we need to use CSS template strings. It seems to work fine with the JavaScript object syntax:

1. sc: https://codesandbox.io/s/sparkling-snowflake-e3cfl?file=/src/App.js
2. emotion: https://codesandbox.io/s/optimistic-ellis-fv65u?file=/src/App.js

Reply:
Ok, weird, it depends on the props to be present